lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -168,6 +168,267 @@ class ConvertAtenEmbeddingOp : public OpConversionPattern<AtenEmbeddingOp> {
+                          castIndexToInt64(rewriter, loc, offsetsLength),
+                          offsetIndexPlusOneInt);
+                      Value nextOffset = rewriter.create<arith::SelectOp>(
+                          loc, checkLast,
+                          castIndexToInt64(rewriter, loc, indicesLength),
+                          rewriter.create<tensor::ExtractOp>(loc, offsets,
+                                                            offsetIndexPlusOne));
+
+                      Value indicesIndex = castIndexToInt64(
+                          rewriter, loc,
+                          rewriter.create<linalg::IndexOp>(loc, /*value=*/1));
+
+                      Value offsetLessThanIndicesIndex =
+                          rewriter.create<arith::CmpIOp>(
+                              loc, arith::CmpIPredicate::slt, offsetsI,
+                              indicesIndex);
+                      Value offsetEqualToIndicesIndex =
+                          rewriter.create<arith::CmpIOp>(loc,
+                                                        arith::CmpIPredicate::eq,
+                                                        offsetsI, indicesIndex);
+                      Value offsetLessThanOrEqualToIndicesIndex =
+                          rewriter.create<arith::OrIOp>(
+                              loc, offsetLessThanIndicesIndex,
+                              offsetEqualToIndicesIndex);
+
+                      Value indicesIndexLessThanNextOffset =
+                          rewriter.create<arith::CmpIOp>(
+                              loc, arith::CmpIPredicate::slt, indicesIndex,
+                              nextOffset);
+
+                      Value indicesIndexWithinBounds =
+                          rewriter.create<arith::AndIOp>(
+                              loc, offsetLessThanOrEqualToIndicesIndex,
+                              indicesIndexLessThanNextOffset);
+
+                      SmallVector<Value> index_into_weight;
+                      index_into_weight.push_back(
+                          castIntToIndex(rewriter, loc, indexInIndices));
+                      index_into_weight.push_back(
+                          rewriter.create<linalg::IndexOp>(loc, /*value=*/2));
+                      Value weightElem = rewriter.create<tensor::ExtractOp>(
+                          loc, weight, index_into_weight);
+
+                      Value addResult = rewriter.create<arith::AddFOp>(
+                          loc, weightElem, initTensorElem);
+                      Value select = rewriter.create<arith::SelectOp>(
+                          loc, indicesIndexWithinBounds, addResult,
+                          initTensorElem);
+                      rewriter.create<linalg::YieldOp>(loc, select);
+                    })
+                .getResult(0);
+
+        // cast outputType.
+        auto resultType1 = typeConverter->convertType(op->getResult(0).getType());
```

Comment:
nit: can you number starting from `0` so that it matches the indexing used in `op->getResult(..)`?